common/tier4_autoware_utils/src/vehicle/vehicle_state_checker.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "tier4_autoware_utils/vehicle/vehicle_state_checker.hpp"
+
+#include "tier4_autoware_utils/trajectory/trajectory.hpp"
+
+#include <string>
+
+namespace tier4_autoware_utils
+{
+VehicleStopChecker::VehicleStopChecker(rclcpp::Node * node)
+: clock_(node->get_clock()), logger_(node->get_logger())
+{
+  using std::placeholders::_1;
+
+  sub_odom_ = node->create_subscription<Odometry>(
+    "/localization/kinematic_state", rclcpp::QoS(1),
+    std::bind(&VehicleStopChecker::onOdom, this, _1));
+}
+
+bool VehicleStopChecker::isVehicleStopped(const double stop_duration = 0.0) const
+{
+  if (twist_buffer_.empty()) {
+    return false;
+  }
+
+  const auto now = clock_->now();
+  const auto time_buffer = now - twist_buffer_.back().header.stamp;
+  if (time_buffer.seconds() < stop_duration) {
+    return false;
+  }
+
+  // Get velocities within stop_duration
+  std::vector<double> vs;
```

Comment:
Why are new vector is created? I think velocity checking can be move to the same loop.

Reply:
Fix in https://github.com/autowarefoundation/autoware.universe/pull/896/commits/f09a0f856e1ebf7f8ac121dd917a57c175ac0c75.